File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        
        Stack opened = new Stack();
        
        for (int i=0; i<s.length(); i++) {
            if (s.substring(i, i+1).equals("(")) {
                opened.push(1);
            } else {
                if (opened.empty()) {
                    return false;
                }
                opened.pop();
            }
        }
        
        // System.out.println(Arrays.toString(opened.toArray()));
        
        if (opened.toArray().length == 0) {
            return true;
        }

        return false;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    s = list(s)
    o = []
    if (s[0] != '(') or (s[-1] != ')'):
        return False
    for i in range(len(s)):
        if s[i] == '(':
            o.append(i)
        else:
            if len(o) == 0:
                return False
            o.pop()
    
    if len(o) == 0:
        return True
    return False